python/llama-index-server/llama_index/server/api/routers/chat.py

```
@@ -95,6 +101,21 @@ async def chat(
+    @router.post("/file")
+    async def upload_file(request: ChatFile) -> ServerFile:
+        """
+        Upload a file to the server to be used in the chat session.
+        """
+        try:
+            save_dir = os.path.join("output", "private")
+            content, _ = FileService._preprocess_base64_file(request.base64)
+            file = FileService.save_file(content, request.name, save_dir)
+            return file
+        except Exception:
+            raise HTTPException(status_code=500, detail="Error uploading file")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)